### PR TITLE
Add a --model-default option to bootstrap.

### PR DESCRIPTION
**Pull request by howbazaar**
This addresses http://pad.lv/1628999 to allow specifying model-defaults on the command line during bootstrap. Previously, only config values specified in the cloud definition would be inherited by the new models in the controller.

### CODE REVIEW

**Comment by howbazaar** [comment]
I'd like to talk to @wallyworld in particular about this change, and some QA steps coming later.

File: cmd/juju/commands/bootstrap.go [hunk 1]
```diff
@@ -174,6 +175,7 @@ func (c *bootstrapCommand) SetFlags(f *gnuflag.FlagSet) {
 	f.StringVar(&c.AgentVersionParam, "agent-version", "", "Version of tools to use for Juju agents")
 	f.StringVar(&c.CredentialName, "credential", "", "Credentials to use when bootstrapping")
 	f.Var(&c.config, "config", "Specify a controller configuration file, or one or more configuration\n    options\n    (--config config.yaml [--config key=value ...])")
+	f.Var(&c.modelDefaults, "model-default", "Specify a configuration file, or one or more configuration\n    options to be set for all models\n    (--config config.yaml [--config key=value ...])")
```
Comment (wallyworld) [inline cmd/juju/commands/bootstrap.go]: ... to be set for all models, unless otherwise specified.

File: cmd/juju/commands/bootstrap.go [hunk 2]
```diff
@@ -536,6 +552,11 @@ func (c *bootstrapCommand) Run(ctx *cmd.Context) (resultErr error) {
 		}
 		inheritedControllerAttrs[k] = v
 	}
+	// Model defaults are added to the inherited controller attributes.
+	// Any command line set model defaults override what is in the cloud config.
+	for k, v := range modelDefaultConfigAttrs {
```
Comment (wallyworld) [inline cmd/juju/commands/bootstrap.go]: We should check and error if they try and specify a bootstrap or controller only attribute in the model defaults. Maybe easiest to use a switch{} inside a for{} similar to above
(and a test)

**Comment by howbazaar** [comment]
@wallyworld updated, please approve if good.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**